src/index.js

```diff
@@ -51,6 +53,11 @@ function resultContainer() {
   }
 }
 
+function cleanup() {
+  unmounts.forEach((unmount) => unmount())
+  unmounts.length = 0
```

Comment:
is there significant difference between this and `unmounts = []`?  That feels a bit clearer what it's doing to me, but I'm not sure if there are some other hidden benefits to this approach?

Reply:
I was thinking to use either mutable or immutable approach. Personally, i prefer the immutable way as i suggested in the issue.

```
let unmounts = [];
unmounts = [...unmounts, unmount];
unmounts = [];
```

but i saw the way that you are using the `resolvers`

```
const resolvers = []; 
resolvers.push(resolver);
``` 
and i tried to follow the same pattern for consistency (mutable). I don't have a strong opinion on this, it's up to you.